src/main/java/io/nats/client/impl/NatsConnection.java

```diff
@@ -553,9 +553,6 @@ public void run() {
                 statusLock.unlock();
             }
             timeTraceLogger.trace("status updated");
-        } catch (RuntimeException exp) { // runtime exceptions, like illegalArgs
-            processException(exp);
-            throw exp;

```

Comment:
This is simply adding the ability to report this message via process exception, which is why it is caught and rethrown. We can revisit this in a different PR if you still feel strongly about this.